conpr/08_LE_Tasks/src/le/fj/Sum.java

```diff
@@ -2,22 +2,26 @@
 
 import java.util.Random;
 import java.util.concurrent.ForkJoinPool;
-import java.util.concurrent.RecursiveTask;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Callable;
 
 @SuppressWarnings("serial")
-public class Sum extends RecursiveTask<Integer> {
+public class Sum implements Callable<Integer> {
     final int THRESHOLD = 1000000;
     final int[] array;
     final int lo;
     final int hi;
+    final ExecutorService exec;
 
-    Sum(int[] array, int lo, int hi) {
+    Sum(int[] array, int lo, int hi, ExecutorService exec) {
         this.array = array;
         this.lo = lo;
         this.hi = hi;
+        this.exec = exec;
     }
 
-    protected Integer compute() {
+    public Integer call() {
```

Comment:
Previously this was the implementation of the abstract method [RecursiveTask.compute()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/RecursiveTask.html#compute()), which returns the result of the computation performed by the task which this file's class (Sum) represents. The method in Callable that does this is [Callable.call()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/Callable.html#call()), which is public (rather than protected like RecursiveTask.compute()).